### PR TITLE
MS-1592 API end points for save report tokens

### PR DESCRIPTION
**Pull request by sulthan-ahmed**
* These tokens are for email links
* GET - using uuid
* POST - providing created date, uuid & email

### CODE REVIEW

File: models/knex/migrations/20200302173446_data-service-migration.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable filenames/match-regex */
+'use strict';
+
+module.exports.up = (knex) => {
+ return knex.schema.createTable('short_life_tokens', (tableReference) => {
```
Comment (jjrobothamho) [inline models/knex/migrations/20200302173446_data-service-migration.js]: How do we use short life tokens?

File: lib/routes/save-report-tokens.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+   debugDB(`Get request data: ${util.inspect(data)}`);
+
+   try {
+     const result = await dataService.write(data, tableName);
+
+     res.json(result);
+     next(false);
+   } catch (err) {
+    next(new Error(`request could not be processed:, ${err.message}`));
```
Comment (jjrobothamho) [inline lib/routes/save-report-tokens.js]: cool looks good

File: models/knex/migrations/20200302173446_data-service-migration.js (hunk 1 above)
Comment (sulthan-ahmed) [reply models/knex/migrations/20200302173446_data-service-migration.js]: this file should be deleted @jjrobothamho thanks for highlighting

Comment (sulthan-ahmed) [reply models/knex/migrations/20200302173446_data-service-migration.js]: i can't find the file so I'll merge and then delete it in another pr

**Pull request merged**